pkg/app/app.go

```diff
@@ -125,6 +126,33 @@ func defaultInputDirectory() configuration.DefaultValueFunction {
 	return callback
 }
 
+func defaultTempDirectory(engine workflow.Engine, config configuration.Configuration, logger *zerolog.Logger) configuration.DefaultValueFunction {
+	callback := func(existingValue interface{}) interface{} {
+		if existingValue != nil {
+			return existingValue
+		}
+		ri := engine.GetRuntimeInfo()
```

Comment:
Suggestion: Let's combine the case where version is empty and runtimeinfo is nil.